gobblin-api/src/main/java/org/apache/gobblin/dataset/DatasetDescriptor.java

```diff
@@ -17,34 +17,52 @@
 
 package org.apache.gobblin.dataset;
 
-import java.util.Map;
-
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
-
 import lombok.Getter;
 
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Objects;
+
 
 /**
  * A {@link Descriptor} identifies and provides metadata to describe a dataset
  */
 public class DatasetDescriptor extends Descriptor {
```

Comment:
One comment: it is not clear from this class if it was intended to capture the notion of a Dataset instance. If so, there might be advantages to extending this class to a new descriptor that defines a dataset instance where the "clusterName" is mandatory.